### PR TITLE
[redis] Add the auth_username_ to health checks

### PR DESCRIPTION
**Pull request by jamesbibby**
Commit Message: Include the auth_username when doing a redis healthcheck (supports Redis 6 ACL Auth for healthchecks)

Additional Description:  I was mistaken in my previous PR thinking this was not possible.   This should enable usernames (Redis 6 ACL) in health checks as well

### CODE REVIEW

**Comment by jamesbibby** [comment]
🤦  I left this out of the previous PR due to a lack familiarity with the protogen process, this should fixup healthchecks for redis so they support the ACL username as well as password.

Fixes #11736

**Comment by asraa** [comment]
Thanks! @jamesbibby can you merge master please to fix timeout?

**Comment by mattklein123** [comment]
Please also add a test that covers this case. Thank you!

/wait

**Comment by jamesbibby** [comment]
> Thanks!

Its not quite done yet, I pushed up the partial test so I can review with @HenryYYang and get it complete.

**Comment by mattklein123** [comment]
OK sounds good will wait for @HenryYYang to review.

/wait

**Comment by jamesbibby** [comment]
Huge thanks to @HenryYYang who helped me through the tests.   This is ready for another review.

**Comment by jamesbibby** [comment]
@mattklein123 quick ping to take a look if you get a chance.

File: test/extensions/health_checkers/redis/redis_test.cc [hunk 1]
```diff
@@ -215,8 +290,108 @@ class RedisHealthCheckerTest
...
+TEST_F(RedisHealthCheckerTest, PingWithAuth) {
+  InSequence s;
+
+  auth_username_ = "test user";
+  auth_password_ = "test password";
+
+  setupWithAuth();
+
+  // Exercise stubbed out interfaces for coverage.
+  exerciseStubs();
```
Comment (HenryYYang) [inline test/extensions/health_checkers/redis/redis_test.cc:306]: I'm not sure if this is needed.

Comment (jamesbibby) [reply test/extensions/health_checkers/redis/redis_test.cc]: Removed thanks

**Comment by jamesbibby** [comment]
There was a small issue with the merge, fixed now and the tests should run. @HenryYYang @mattklein123

**Comment by HenryYYang** [comment]
LGTM, thanks James

**Pull request merged**